src/beamlime/applications/_nexus_helpers.py

```diff
@@ -1,155 +1,360 @@
+
+    It was named as a group, not a module
+    since the module place holder is one of the children of it.
+    """
+
+    children: list[Mapping]
+    """A module place holder or datasets."""
+    name: str
+    """Name of the group."""
+
+
+NexusGroupDictStore = dict[NexusPath, NexusGroupDict]
+"""A dictionary to store nexus groups for merging data pieces."""
+NexusDictStructure = dict
+"""A whole/partial nexus structure used for the data acquisition."""
+
+
+def create_dataset(
+    *, name: str, dtype: str, initial_values: Any, unit: str | None = None
+) -> NexusDatasetDict:
+    """Creates a dataset according to the arguments."""
+    dataset: NexusDatasetDict = {
+        "module": "dataset",
+        "config": {
+            "name": name,
+            "dtype": dtype,
+            "values": initial_values,
+        },
+        "attributes": [],
+    }
+    if unit is not None:
+        attrs = dataset["attributes"]
+        attrs.append(
+            {
+                "name": "units",
+                "dtype": "string",
+                "values": unit,
+            }
         )
-        group['children'].append(dataset)
     return dataset
 
 
-def _merge_ev44(group, message):
-    for field, value in message.items():
-        if value is None:  # Could be monitor without pixel_id
-            continue
-        dataset = _init_dataset(group, field, value)
-        if field in ('time_of_flight', 'pixel_id'):
-            dataset['config']['values'] = np.concatenate(
-                (dataset['config']['values'], value)
-            )
+ModuleNameType = Literal["ev44"]  # "f144", "tdct" will be added in the future
+"""Name of the module that is supported by beamlime."""
+
+DeserializedMessage = Mapping
+"""Deserialized message from one of the schemas bound to :attr:`~ModuleNameType`."""
+
+
+def _initialize_ev44(group: NexusGroupDict, data_piece: DeserializedMessage) -> None:
+    """Initialize ev44 datasets in the parent.
 
+    Params
+    ------
+    group:
+        A data group that has a module place holder as a child.
+    data_piece:
+        A deserialized message corresponding to ``ev44`` schema.
 
-def _merge_f144(group, message):
-    for field, value in message.items():
-        dataset = _init_dataset(group, field, value)
-        if field in ('timestamp', 'value'):
-            dataset['config']['values'] = np.stack((dataset['config']['values'], value))
+    Side Effects
+    ------------
+    ``children`` of the ``group`` will have 4 datasets with empty values.
 
+    - event_time_zero
+    - event_time_offset
+    - event_index
+    - event_id
 
-def _merge_tdct(group, message):
-    for field, value in message.items():
-        dataset = _init_dataset(group, field, value)
-        if field in ('timestamps',):
-            dataset['config']['values'] = np.concatenate(
-                (dataset['config']['values'], value)
+    """
+    children = group.setdefault("children", [])
```

Comment:
Like I mentioned earlier I think it's better to not create the `children` here if we are already doing it elsewhere.
Doing it in two places creates confusion about whose responsibility it is to initialize it. It's better that we explicitly do not create it here, and if it isn't created elsewhere then this merge routine will just fail.

Reply:
Hmm yeah. It won't create a children if it exists but might be still confusing.

Reply:
I fixed it...!